socorro/unittest/collector/test_wsgi_breakpad_collector.py

```diff
@@ -495,3 +504,490 @@ def test_no_x00_character(self):
+        self,
+        mocked_web,
+        mocked_webapi,
+        mocked_utc_now,
+        mocked_time
+    ):
+        config = self.get_standard_config()
+        c = BreakpadCollector2015(config)
+        rawform = DotDict()
+        rawform.ProductName = 'FireSquid'
+        rawform.Version = '99'
+        rawform.dump = DotDict({'value': 'fake dump', 'file': 'faked file'})
+        rawform.aux_dump = DotDict({'value': 'aux_dump contents', 'file': 'silliness'})
+        rawform.some_field = '23'
+        rawform.some_other_field = ObjectWithValue('XYZ')
+        rawform.uuid = '332d798f-3c42-47a5-843f-a0f892140107'
+        rawform.legacy_processing = u'1'
+
+        form = DotDict(rawform)
+        form.dump = rawform.dump.value
+
+        erc = DotDict()
+        erc.ProductName = 'FireSquid'
+        erc.Version = '99'
+        erc.some_field = '23'
+        erc.some_other_field = 'XYZ'
+        erc.legacy_processing = ACCEPT
+        erc.timestamp = 3.0
+        erc.submitted_timestamp = '2012-05-04T15:10:00'
+        erc.throttle_rate = 100
+        erc.dump_checksums = {
+            'dump': '2036fd064f93a0d086cf236c5f0fd8d4',
+            'aux_dump': 'aa2e5bf71df8a4730446b2551d29cb3a',
+        }
+        erc.type_tag = 'bp'
+        erc = dict(erc)
+
+        mocked_web.input.return_value = form
+        mocked_webapi.rawinput.return_value = rawform
+        mocked_utc_now.return_value = datetime(2012, 5, 4, 15, 10)
+        mocked_time.time.return_value = 3.0
+        c.throttler.throttle.return_value = (ACCEPT, 100)
+        r = c.POST()
+        ok_(r.startswith('CrashID=bp-'))
+        ok_(r.endswith('120504\n'))
+        erc['uuid'] = r[11:-1]
+        c.crash_storage.save_raw_crash.assert_called_with(
+            erc,
+            {'dump':'fake dump', 'aux_dump':'aux_dump contents'},
+            r[11:-1]
+        )
+
+    @mock.patch('socorro.collector.wsgi_breakpad_collector.time')
+    @mock.patch('socorro.collector.wsgi_breakpad_collector.utc_now')
+    @mock.patch('socorro.collector.wsgi_generic_collector.web.webapi')
+    @mock.patch('socorro.collector.wsgi_generic_collector.web')
+    def test_POST_with_existing_legacy_processing_and_use_it(
+        self,
+        mocked_web,
+        mocked_webapi,
+        mocked_utc_now,
+        mocked_time
+    ):
+        config = self.get_standard_config()
+        config.accept_submitted_crash_id = True
+        config.accept_submitted_legacy_processing = True
+        c = BreakpadCollector2015(config)
+
+        rawform = DotDict()
+        rawform.ProductName = 'FireSquid'
+        rawform.Version = '99\x00'
+        rawform.dump = DotDict({'value': 'fake dump', 'file': 'faked file'})
+        rawform.aux_dump = DotDict({'value': 'aux_dump contents', 'file': 'silliness'})
+        rawform[u'some_field\u0000'] = '23'
+        rawform[u'some_\u0000other_field'] = ObjectWithValue('XYZ')
+        rawform.uuid = '332d798f-3c42-47a5-843f-a0f892140107'
```

Comment:
You seem to be repeating those lines over and over in your tests. Could you put them into a method instead?